src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -756,6 +773,16 @@ String getAppPath(String appName) {
         return Arrays.asList("-o", getAppPath(appName));
     }
 
+    protected List<String> getNativeCodeList() throws IOException {
+        Path nativeCodeDir = paths.getNativeCodePath();
+        if (!Files.exists(nativeCodeDir))
+            return Collections.emptyList();
+        return Files.list(nativeCodeDir)
+            .map(p -> p.getFileName().toString())
```

Comment:
check non null objects, non null file name, if that make sense?

Reply:
`paths.getNativeCodePath()` always returns non null value (since it only appends `native` on project path)
Directory is checked on line 778, so `Files.list(nativeCodeDir)` will be either `Stream` or empty `Stream`.

Reply:
ok